src/imphint.d

```diff
@@ -19,35 +19,24 @@ import core.stdc.string;
  */
 extern (C++) const(char)* importHint(const(char)* s)
 {
-    static __gshared const(char)** modules = ["core.stdc.stdio", "std.stdio", "std.math", "core.vararg", null];
-    static __gshared const(char)** names =
-    [
-        "printf",
-        null,
-        "writeln",
-        null,
-        "sin",
-        "cos",
-        "sqrt",
-        "fabs",
-        null,
-        "__va_argsave_t",
-        null
+    return hints.get(s[0..strlen(s)], null);
+}
+
+private immutable char*[const(char)[]] hints;
```

Comment:
Can you make this `immutable string[string]`? It looks simpler and practically works the same way. I think you only need to add `.ptr` at the end of the return statement in `importHint`, because `get` will return a slice which (even when it is `null`) has a `.ptr` member.